paper-server/src/main/java/com/destroystokyo/paper/console/PaperConsole.java

```diff
@@ -42,7 +46,29 @@ protected boolean isRunning() {
 
     @Override
     protected void runCommand(String command) {
-        this.server.handleConsoleInput(command, this.server.createCommandSourceStack());
+        // terminals interpret pressing [enter] and pasting a multi-line string differently,
+        // the latter makes the line reader read it as a single line - and we don't want that
+        // https://github.com/PaperMC/Paper/issues/13006
+        outer:
+        for (String line : command.split("\n")) {
```

Comment:
Not 100% sure about how the terminal on different systems handles line breaks, but I think this should be using command.lines()?

Reply:
Forgot about `.lines()`, fixed now (though line breaks are indeed already normalized to LF here)